[RM Ch08] Single Pane of Glass (#2902)
cntt-n/CNTT: doc/ref_model/chapters/chapter08.rst

-  component clouds only surface NBIs (Please note that VMware deployed in a private and a public cloud can be treated as a single cloud instance)

Characteristics of a Federated Cloud
------------------------------------

> In this section we will further explore the characteristics of the federated cloud architecture, and architecture building blocks that constitute the federated cloud. For example, :numref:`Example Hybrid Multi-Cloud Component Cloud` shows a Telco Cloud that consists of 4 sub-clouds: Private on premise, Cloud Vendor provided on premise, Private outsourced (Commercial Cloud Provider such as a Hyperscaler Cloud Provider (HCP), and Public outsourced (see diagram below). Such an implementation of a Telco Cloud allows for mix'n'match of price points, flexibility in market positioning and time to market, capacity with the objective of attaining near "unlimited" capacity, scaling within a sub-cloud or through bursting across sub-clouds, access to "local" capacity near user base, and access to specialised services.

Author: ```suggestion
In this section, we will further explore the characteristics of the federated cloud architecture, and architecture building 
blocks that constitute the federated cloud. For example, :numref:`Example Hybrid Multi-Cloud Component Cloud` 
shows a Telco Cloud that consists of 4 sub-clouds: Private on premise, Cloud Vendor provided on premise, Private 
outsourced (Commercial Cloud Provider such as a Hyperscaler Cloud Provider (HCP), and Public outsourced (see 
diagram below). Such an implementation of a Telco Cloud allows for mix'n'match of price points, flexibility in market 
positioning and time to market, capacity with the objective of attaining near "unlimited" capacity, scaling within a sub-cloud or through bursting across sub-clouds, access to "local" capacity near user base, and access to specialised 
services.
```

-  component clouds only surface NBIs (Please note that VMware deployed in a private and a public cloud can be treated as a single cloud instance)

Characteristics of a Federated Cloud
------------------------------------

In this section, we will further explore the characteristics of the federated cloud architecture, and architecture building 
blocks that constitute the federated cloud. For example, :numref:`Example Hybrid Multi-Cloud Component Cloud` 
shows a Telco Cloud that consists of 4 sub-clouds: Private on premise, Cloud Vendor provided on premise, Private 
outsourced (Commercial Cloud Provider such as a Hyperscaler Cloud Provider (HCP), and Public outsourced (see 
diagram below). Such an implementation of a Telco Cloud allows for mix'n'match of price points, flexibility in market 

> positioning and time to market, capacity with the objective of attaining near "unlimited" capacity, scaling within a sub-cloud or through bursting across sub-clouds, access to "local" capacity near user base, and access to specialised 

Author: ```suggestion
positioning and time to market, capacity with the objective of attaining near "unlimited" capacity, scaling within a 
sub-cloud or through bursting across sub-clouds, access to "local" capacity near user base, and access to specialised 
```

   Example Hybrid Multi-Cloud Component Cloud

Telco Cloud
-----------

> The :numref:`Telco Cloud: Data Centre to Edge` presents a visualisation of a Telco operator cloud (or simply, Telco cloud) with clouds and cloud components distributed across Regional Data Centres, Metro locations (such as Central Office or a Colocation site) and at the Edge, that are interconnected using a partial mesh network. Please note that at the Regional centre level the interconnections are likely to be a "fuller" mesh while being a sparser mesh at the Edges.

Author: ```suggestion
The :numref:`Telco Cloud: Data Centre to Edge` presents a visualisation of a Telco operator cloud (or simply, Telco 
cloud) with clouds and cloud components distributed across Regional Data Centres, Metro locations (such as Central 
Office or a Colocation site) and at the Edge, that are interconnected using a partial mesh network. Please note that at 
the Regional centre level the interconnections are likely to be a "fuller" mesh while being a sparser mesh at the Edges.
```

> The Cloud Services and the Cloud Resources Brokers provide value-added services in addition to the fundamental capabilities like service and resource discovery. These Brokers are critical for a multi-cloud environment to function and utilise cloud specific plugins to perform the necessary activities. These Brokers can, for example, provision and manage environments with resources and services for Machine Learning (ML) services, Augmented/Virtual Reality, or specific industries.

Author: ```suggestion
The Cloud Services and the Cloud Resources Brokers provide value-added services in addition to the fundamental 
capabilities like service and resource discovery. These Brokers are critical for a multi-cloud environment to function 
and utilise cloud specific plugins to perform the necessary activities. These Brokers can, for example, provision and 
manage environments with resources and services for Machine Learning (ML) services, Augmented/Virtual Reality, or 
specific industries.
```

Requirements, Reference Architecture & Industry Standards Intersect
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

The Communcations Service Provider (CSP) is both a provider and consumer of Cloud based services.
When the CSP is actings as:

-  consumer, in which case the typical consideration is total cost of ownership as the consumption is to usually to support internal business operations: BSS/OSS systems;
-  provider of cloud services, through operation of their own cloud or reselling of cloud services, in which case the typical consideration is margin (cost to offer services vs income received).

These two stances will drive differing approaches to how a CSP would look to manage how it interacts within a Multi-Cloud environment.

> Single Pane of Glass

Author: ```suggestion
Unified Management of Hybrid, Edge, and Multi-Cloud Environments
```

> Unified Management of Hybrid, Edge, and Multi-Cloud Environments

Author: ```suggestion
Hybrid, Edge, and Multi-Cloud unified management Platform 
```

~~~~~~~~~~~~~~~~~~~~

> As organisations spread their resources across on-premises, multiple clouds, and the Edge, the need for a single set of tools and processes to manage and operate across these Hybrid, Edge, and Multi-clouds (HEM clouds) is obvious as can be seen form the following simplistic sceanrio. 

Reviewer: @pgoyal01 Typos: form -> from, sceanrio -> scenario

Author: @karinesevilla Thanks - Done. Please review and resolve conversation.

> As organisations spread their resources across on-premises, multiple clouds, and the Edge, the need for a single set of tools and processes to manage and operate across these Hybrid, Edge, and Multi-clouds (HEM clouds) is obvious as can be seen form the following simplistic sceanrio. 

Reviewer: @pgoyal01 Typos: form -> from, sceanrio -> scenario

> As organisations spread their resources across on-premises, multiple clouds, and the Edge, the need for a single set of tools and processes to manage and operate across these Hybrid, Edge, and Multi-clouds (HEM clouds) is obvious as can be seen form the following simplistic sceanrio. 

Author: Thanks @karinesevilla 
```suggestion
As organisations spread their resources across on-premises, multiple clouds, and the Edge, the need for a single set of tools and processes to manage and operate across these Hybrid, Edge, and Multi-clouds (HEM clouds) is obvious as can be seen from the following simplistic sceanario. 
```

> As organisations spread their resources across on-premises, multiple clouds, and the Edge, the need for a single set of tools and processes to manage and operate across these Hybrid, Edge, and Multi-clouds (HEM clouds) is obvious as can be seen form the following simplistic sceanrio. 

Author: ```suggestion
As organisations spread their resources across on-premises, multiple clouds, and the Edge, the need for a single set of tools and processes to manage and operate across these Hybrid, Edge, and Multi-clouds (HEM clouds) is obvious as can be seen from the following simplistic sceanrios. 
```

> As organisations spread their resources across on-premises, multiple clouds, and the Edge, the need for a single set of tools and processes to manage and operate across these Hybrid, Edge, and Multi-clouds (HEM clouds) is obvious as can be seen from the following simplistic sceanario. 

Author: ```suggestion
As organisations spread their resources across on-premises, multiple clouds, and the Edge, the need for a single set of tools and processes to manage and operate across these Hybrid, Edge, and Multi-clouds (HEM clouds) is obvious as can be seen from the following simplistic sceanarios. 
```

> Scenario: An opertor has private clouds that it utilises for its workloads.

Reviewer: @pgoyal01 Is it the initial situation, which will be transformed into A, B or C solutions?

Author: @karinesevilla It will evolve -- made changes, please review. Thanks

> Scenario: An opertor has private clouds that it utilises for its workloads.

Reviewer: @pgoyal01 Is it the initial situation, which will be transformed into A, B or C solutions?

> Scenario: An operator has private clouds that it utilises for its workloads.

Author: ```suggestion
Scenario: An operator has private clouds that it utilises for its workloads. Over time, the operator evolves their environment:
```

- A: The operator has decided to utilise one or more public clouds for some of its workloads.

> - B: The operator has decided to utilise an edge clouds for some of its clients.

Reviewer 2: Typo. Change to "utilize an edge cloud", "utilize edge clouds" or "utilize one or more edge clouds".

Author: @ulikleber Thanks - Done; please review and resolve this conversation.

> - B: The operator has decided to utilise an edge clouds for some of its clients.

Author: ```suggestion
- B: The operator has decided to utilise an edge cloud for some of its clients.
```

- C: The operator has decided to create edge clouds for some of its clients.

> Scenario B can be treated being the same as Scenario A. Scenario C is akin to the private cloud except for location and control over the facilities at that location. For its worklooads, the operator will have to utilise the target clouds tools or APIs to create the necessary accounts, billing arrangements, quotas, etc. Then create the needed resources, such as VMs or kubernetes clusters, etc. Following up with creating needed storage, networking, etc. before onboarding the workload and operating it. This is complex even when the operator is delaing with say only one other cloud, in addition to operating its own cloud. The operator is faced with a number of challenges including acquiring a new set of skills, knowledge of APIs, tools, and the complexity of managing different policies, updates, etc. This becomes impossible to manage when incorporating more than one other cloud. Hence the need for a Single Pane of Glass

Author: @ulikleber I hope I am not saying that but rather pointing out that it would be impossible to manage if more than 2 clouds. Let us discuss in RM meeting. Thabks

> Scenario B can be treated being the same as Scenario A. Scenario C is akin to the private cloud except for location and control over the facilities at that location. For its worklooads, the operator will have to utilise the target clouds tools or APIs to create the necessary accounts, billing arrangements, quotas, etc. Then create the needed resources, such as VMs or kubernetes clusters, etc. Following up with creating needed storage, networking, etc. before onboarding the workload and operating it. This is complex even when the operator is delaing with say only one other cloud, in addition to operating its own cloud. The operator is faced with a number of challenges including acquiring a new set of skills, knowledge of APIs, tools, and the complexity of managing different policies, updates, etc. This becomes impossible to manage when incorporating more than one other cloud. Hence the need for a Single Pane of Glass

Author: @ulikleber Thanks
```suggestion
Scenario B can be treated being the same as Scenario A. Scenario C is akin to the private cloud except for location and control over the facilities at that location. For its worklooads, the operator will have to utilise the target clouds tools or APIs to create the necessary accounts, billing arrangements, quotas, etc. Then create the needed resources, such as VMs or kubernetes clusters, etc. Following up with creating needed storage, networking, etc. before onboarding the workload and operating it. This is complex even when the operator is dealing with say only one other cloud, in addition to operating its own cloud. The operator is faced with a number of challenges including acquiring a new set of skills, knowledge of APIs, tools, and the complexity of managing different policies, updates, etc. This becomes impossible to manage when incorporating more than one other cloud. Hence the need for a Single Pane of Glass
```

> This Single-Pane-of-Glass, the HEM operator's Platform (HEMP), provides capabilities to conistently:

Reviewer 2: typo coistently
```suggestion
This Single-Pane-of-Glass, the HEM operator's Platform (HEMP), provides capabilities to consistently:
```

> 

Author: @karinesevilla Thanks
```suggestion

-  manage accounts, credentials, and resources
```

> -  manage Services, resources, etc.

Reviewer: @pgoyal01 I think that "accounts, credentials and resources" should be added before services into this sentence

Author: @karinesevilla Thanks - Done

Reviewer: @pgoyal01 Reading again the list, It could be better to have all in the same sentence: manage accounts, credentials, resources, and services. It was the meaning of my first comment

Author: @karinesevilla Done - Thanks. Please review and resolve

> -  manage Services, resources, etc.

Reviewer: @pgoyal01 I think that "accounts, credentials and resources" should be added before services into this sentence

> This Single-Pane-of-Glass, the HEM operator's Platform (HEMP), provides capabilities to consistently:

Author: ```suggestion
This Hybrid, Edge, and Multi-Cloud unified management Platform (HEMP) (a.k.a. Single-Pane-of-Glass) provides capabilities to consistently:
```

  -  across facilities (regions, data centers, edge locations)
-  interoperate the different clouds

> -  implement comon policies and governance standards

Reviewer 2: typo comon
```suggestion
-  implement common policies and governance standards
```

   
* - Ref
  - Requirement
  - Definition/Note
* - hem.gen.001
  - HEMP shall use only published APIs in managing component clouds. 
  - For example, to accomplish the example in `hem.gen.003` it will use the published APIs of the target cloud.
* - hem.gen.002
  - HEMP shall publish all of the APIs used by any of its components.
  - For example, the provided GUI portal shall only use HEMP published APIs.
* - hem.gen.003
  - HEMP shall provide for a common mechanism for interaction with its constituent clouds.
  - For example, "create Account" across the different public clouds. 
* - hem.gen.004
  - HEMP shall generalise and define a common set of resources available to be managed in constituent clouds.
  - Example resources: hosts (including BareMetal), Virtual Machines (VM), vCPU, Memory, Storage, Network, kubernetes clusters, kubernetes nodes, Images (OS, and others), credentials. For private cloud additional example resources: Racks, ToR/CE switches, Platform images. 

Table : Hybrid, Edge, and Multi cloud operator Platform (HEMP) General Requirements

.. list-table::
   :widths: 16, 30, 30
   :header-rows: 1
   
* - Ref
  - Requirement
  - Definition/Note
* - hem.ops.001

>   - HEMP shall generalise and define a common set of management operations available in constituent clouds. Some operations may only be available for a subset of resources.

Reviewer: This requirement is too vague or imprecise

>   - HEMP shall generalise and define a common set of management operations available in constituent clouds. Some operations may only be available for a subset of resources.

Reviewer: This requirement is too vague or imprecise

  - For example, create, deploy, configure, start, suspend, stop, resume, reboot, delete, scale, list.
* - hem.ops.002
  - HEMP shall centrally manage all resources (across all constituent clouds).
  - 
* - hem.ops.003

>   - HEMP shall cetrally operate all constituent clouds.

Reviewer 2: typo cetrally
```suggestion
  - HEMP shall centrally operate all constituent clouds.
```

  -
* - hem.ops.004
  - HEMP shall provide mechanisms to integrate new clouds.
  - This may require pre-development of necessary capabilities for the support of HEMP abstractions, and impementation of connectivity with the new cloud.
* - hem.ops.005
  - HEMP shall provide mechanisms to drop a constituent cloud.
  - This will result in deletion of all resources in the constituent cloud.
* - hem.ops.006
  - HEMP shall provide mechanisms and processes to onboard existing assets (resources, connectivity, etc.).
  -
* - hem.ops.007
  - HEMP shall provide mechanisms and processes for the automated configuration management of all environments and resources.
  -

Table : Hybrid, Edge, and Multi cloud operator Platform (HEMP) Operability Requirements

.. list-table::
   :widths: 16, 30, 30
   :header-rows: 1
   
* - Ref
  - Requirement
  - Definition/Note
* - hem.lcm.001
  - HEMP shall provide visibility into the health of all assets.
  - 
* - hem.lcm.002

>   - HEMP shall monitor all environments and assets.

Reviewer: The requirement HEM.LCM.002 is the umbrella requirement of all requirements in this table. It can be include in all the other requirements.

Reviewer: Yes, I wonder if HEM.LCM.002 is useful as the other requirements are addressing dedicated aspects of monitoring

Author: @karinesevilla we can delete if you feel strongly about it

Reviewer: @pankaj Instead of deleting it, I would have placed it at first, switching  hem.lcm.001 and hem.lcm.002

  -
* - hem.lcm.003
  - HEMP shall provide capabilities for a centralised visibility and management of all alerts.
  -
* - hem.lcm.004
  - HEMP shall provide capabilities for a centralised analysis of all logs.
  - This doesn't preclude local log analytics.

Table : Hybrid, Edge, and Multi cloud operator Platform (HEMP) Life Cycle Management (LCM) Requirements

.. list-table::
   :widths: 16, 30, 30
   :header-rows: 1
   
* - Ref
  - Requirement
  - Definition/Note
* - hem.sec.001
  - HEMP shall provide capabilities for the centralised management of all security policies.
  - 
* - hem.sec.002

>   - HEMP shall provide capabilities for the centralised tracking of compliance.

Reviewer: Compliance with what? It should be precise

>   - HEMP shall provide capabilities for the centralised tracking of compliance.

Reviewer: Compliance with what? It should be precise

- B: The operator has decided to utilise an edge cloud for some of its clients.
- C: The operator has decided to create edge clouds for some of its clients.

Scenario B can be treated being the same as Scenario A. Scenario C is akin to the private cloud except for location and control over the facilities at that location. For its worklooads, the operator will have to utilise the target clouds tools or APIs to create the necessary accounts, billing arrangements, quotas, etc. Then create the needed resources, such as VMs or kubernetes clusters, etc. Following up with creating needed storage, networking, etc. before onboarding the workload and operating it. This is complex even when the operator is dealing with say only one other cloud, in addition to operating its own cloud. The operator is faced with a number of challenges including acquiring a new set of skills, knowledge of APIs, tools, and the complexity of managing different policies, updates, etc. This becomes impossible to manage when incorporating more than one other cloud. Hence the need for a Single Pane of Glass

This Single-Pane-of-Glass, the HEM operator's Platform (HEMP), provides capabilities to consistently:

> -  manage accounts, credentials, and resources

Author: ```suggestion
-  manage accounts, credentials, resources and services
```

> -  manage Services, resources, etc.

Author: ```suggestion
```

   
* - Ref
  - Requirement
  - Definition/Note
* - hem.gen.001
  - HEMP shall use only published APIs in managing component clouds. 
  - For example, to accomplish the example in `hem.gen.003` it will use the published APIs of the target cloud.
* - hem.gen.002
  - HEMP shall publish all of the APIs used by any of its components.
  - For example, the provided GUI portal shall only use HEMP published APIs.
* - hem.gen.003
  - HEMP shall provide for a common mechanism for interaction with its constituent clouds.
  - For example, "create Account" across the different public clouds. 
* - hem.gen.004
  - HEMP shall generalise and define a common set of resources available to be managed in constituent clouds.
  - Example resources: hosts (including BareMetal), Virtual Machines (VM), vCPU, Memory, Storage, Network, kubernetes clusters, kubernetes nodes, Images (OS, and others), credentials. For private cloud additional example resources: Racks, ToR/CE switches, Platform images. 

Table : Hybrid, Edge, and Multi cloud operator Platform (HEMP) General Requirements

.. list-table::
   :widths: 16, 30, 30
   :header-rows: 1
   
* - Ref
  - Requirement
  - Definition/Note
* - hem.ops.001

>   - HEMP shall generalise and define a common set of management operations available in constituent clouds. Some operations may only be available for a subset of resources.

Author: ```suggestion
  - HEMP shall generalise and define a common set of management operations available in constituent clouds;  required operations include: create, deploy, configure, start, suspend, stop, resume, reboot, delete, scale, list. Some operations may only be available for a subset of resources.
```

  -
* - hem.ops.007
  - HEMP shall provide mechanisms and processes for the automated configuration management of all environments and resources.
  -

Table : Hybrid, Edge, and Multi cloud operator Platform (HEMP) Operability Requirements

.. list-table::
   :widths: 16, 30, 30
   :header-rows: 1
   
* - Ref
  - Requirement
  - Definition/Note
* - hem.lcm.001
  - HEMP shall provide visibility into the health of all assets.
  - 
* - hem.lcm.002
  - HEMP shall monitor all environments and assets.
  -
* - hem.lcm.003
  - HEMP shall provide capabilities for a centralised visibility and management of all alerts.
  -
* - hem.lcm.004
  - HEMP shall provide capabilities for a centralised analysis of all logs.
  - This doesn't preclude local log analytics.

Table : Hybrid, Edge, and Multi cloud operator Platform (HEMP) Life Cycle Management (LCM) Requirements

.. list-table::
   :widths: 16, 30, 30
   :header-rows: 1
   
* - Ref
  - Requirement
  - Definition/Note
* - hem.sec.001
  - HEMP shall provide capabilities for the centralised management of all security policies.
  - 
* - hem.sec.002

>   - HEMP shall provide capabilities for the centralised tracking of compliance.

Author: ```suggestion
  - HEMP shall provide capabilities for the centralised tracking of compliance of all security requirements (:ref:`ref_model/chapters/chapter07:consolidated security requirements`).
```

Requirements, Reference Architecture & Industry Standards Intersect
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

The Communcations Service Provider (CSP) is both a provider and consumer of Cloud based services.
When the CSP is actings as:

-  consumer, in which case the typical consideration is total cost of ownership as the consumption is to usually to support internal business operations: BSS/OSS systems;
-  provider of cloud services, through operation of their own cloud or reselling of cloud services, in which case the typical consideration is margin (cost to offer services vs income received).

These two stances will drive differing approaches to how a CSP would look to manage how it interacts within a Multi-Cloud environment.

Hybrid, Edge, and Multi-Cloud unified management Platform 
~~~~~~~~~~~~~~~~~~~~

> As organisations spread their resources across on-premises, multiple clouds, and the Edge, the need for a single set of tools and processes to manage and operate across these Hybrid, Edge, and Multi-clouds (HEM clouds) is obvious as can be seen from the following simplistic sceanarios. 

Reviewer: @pgoyal01 Typo -> scenarios

> As organisations spread their resources across on-premises, multiple clouds, and the Edge, the need for a single set of tools and processes to manage and operate across these Hybrid, Edge, and Multi-clouds (HEM clouds) is obvious as can be seen from the following simplistic sceanarios. 

Reviewer: Linewrap must be 120 chars

> As organisations spread their resources across on-premises, multiple clouds, and the Edge, the need for a single set of tools and processes to manage and operate across these Hybrid, Edge, and Multi-clouds (HEM clouds) is obvious as can be seen from the following simplistic sceanarios. 

Author: ```suggestion
As organisations spread their resources across on-premises, multiple clouds, and the Edge, the need for a single set
of tools and processes to manage and operate across these Hybrid, Edge, and Multi-clouds (HEM clouds) is obvious 
as can be seen from the following simplistic scenarios. 
```

Scenario: An operator has private clouds that it utilises for its workloads. Over time, the operator evolves their environment:

- A: The operator has decided to utilise one or more public clouds for some of its workloads.
- B: The operator has decided to utilise an edge cloud for some of its clients.
- C: The operator has decided to create edge clouds for some of its clients.

> Scenario B can be treated being the same as Scenario A. Scenario C is akin to the private cloud except for location and control over the facilities at that location. For its worklooads, the operator will have to utilise the target clouds tools or APIs to create the necessary accounts, billing arrangements, quotas, etc. Then create the needed resources, such as VMs or kubernetes clusters, etc. Following up with creating needed storage, networking, etc. before onboarding the workload and operating it. This is complex even when the operator is dealing with say only one other cloud, in addition to operating its own cloud. The operator is faced with a number of challenges including acquiring a new set of skills, knowledge of APIs, tools, and the complexity of managing different policies, updates, etc. This becomes impossible to manage when incorporating more than one other cloud. Hence the need for a Single Pane of Glass.

Reviewer: typo: worklooads ->workloads

GSMA's Operator Platform Group (OPG) specify a federated model and specify requirements for the Edge Platforms (Operator Platform Telco Edge Requirements v2.0 (:ref:`https://infocentre2.gsma.com/gp/pr/FNW/OPG/OfficialDocuments/OPG.02%20Operator%20Platform%20Telco%20Edge%20Requirements%20v2.0%20(Current)/OPG.02%20v2.0.pdf`); while the document is for Edge, most of the requirements are easily applicable to other cloud deployments. Anuket RM is implementation agnostic, viz., whether the implementation uses agents, federations or some other mechanisms. 

The following tables list some of the requirements for the Hybrid, Edge, and Multi cloud operator Platform (HEMP). These requirements are in addition to the requirements in other chapters of this RM. 

**HEMP General Requirements**

+-------------+--------------------------------------------------------+------------------------------------------------------------+
| Ref         | Requirement                                            | Definition/Note                                            |
+=============+========================================================+============================================================+

> | hem.gen.001 | HEMP shall use only published APIs in managing         | For example, to accomplish the example in `hem.gen.003`    |

Reviewer: @pgoyal01 I think that it is the first time we are using "shall" rather than "must" to write requirements. Is there a reason for that?

Author: @karinesevilla Thanks -fixing it now

> |             | component clouds				       | it will use the published APIs of the target cloud.        |

Reviewer: Issues on some of this table lines: lack of spaces

|             | and resources                                          |                                                            |
+-------------+--------------------------------------------------------+------------------------------------------------------------+

Table : Hybrid, Edge, and Multi cloud operator Platform (HEMP) Operability Requirements

**HEMP LCM Requirements**

+-------------+--------------------------------------------------------+------------------------------------------------------------+
| Ref         | Requirement                                            | Definition/Note                                            |
+=============+========================================================+============================================================+
| hem.lcm.001 | HEMP shall monitor all environments and assets         |                                                            |
+-------------+--------------------------------------------------------+------------------------------------------------------------+
| hem.lcm.002 | HEMP shall provide visibility into the health of all   |                                                            |

> |             | assets           				       |                                                            |

Reviewer: Issue on table line: lack of spaces

-  provider of cloud services, through operation of their own cloud or reselling of cloud services, in which case the typical consideration is margin (cost to offer services vs income received).

These two stances will drive differing approaches to how a CSP would look to manage how it interacts within a Multi-Cloud environment.

Hybrid, Edge, and Multi-Cloud unified management Platform 
~~~~~~~~~~~~~~~~~~~~

As organisations spread their resources across on-premises, multiple clouds, and the Edge, the need for a single set
of tools and processes to manage and operate across these Hybrid, Edge, and Multi-clouds (HEM clouds) is obvious 
as can be seen from the following simplistic scenarios. 

Scenario: An operator has private clouds that it utilises for its workloads. Over time, the operator evolves their environment:

- A: The operator has decided to utilise one or more public clouds for some of its workloads.
- B: The operator has decided to utilise an edge cloud for some of its clients.
- C: The operator has decided to create edge clouds for some of its clients.

> Scenario B can be treated being the same as Scenario A. Scenario C is akin to the private cloud except for location and control over the facilities at that location. For its worklooads, the operator will have to utilise the target clouds tools or APIs to create the necessary accounts, billing arrangements, quotas, etc. Then create the needed resources, such as VMs or kubernetes clusters, etc. Following up with creating needed storage, networking, etc. before onboarding the workload and operating it. This is complex even when the operator is dealing with say only one other cloud, in addition to operating its own cloud. The operator is faced with a number of challenges including acquiring a new set of skills, knowledge of APIs, tools, and the complexity of managing different policies, updates, etc. This becomes impossible to manage when incorporating more than one other cloud. Hence the need for a Single Pane of Glass.

Author: ```suggestion
Scenario B can be treated as being the same as Scenario A. Scenario C is akin to the private cloud except for location
and control over the facilities at that location. For its workloads, the operator will have to utilise the target clouds 
tools or APIs to create the necessary accounts, billing arrangements, quotas, etc. Then create the needed resources, 
such as VMs or Kubernetes clusters, etc. Following up with creating needed storage, networking, etc. before 
onboarding the workload and operating it. This is complex even when the operator is dealing with say only one other 
cloud, in addition to operating its own cloud. The operator is faced with a number of challenges including acquiring a 
new set of skills, knowledge of APIs, tools, and the complexity of managing different policies, updates, etc. This 
becomes impossible to manage when incorporating more than one other cloud. Hence the need for a Single Pane of Glass.
```

GSMA's Operator Platform Group (OPG) specify a federated model and specify requirements for the Edge Platforms (Operator Platform Telco Edge Requirements v2.0 (:ref:`https://infocentre2.gsma.com/gp/pr/FNW/OPG/OfficialDocuments/OPG.02%20Operator%20Platform%20Telco%20Edge%20Requirements%20v2.0%20(Current)/OPG.02%20v2.0.pdf`); while the document is for Edge, most of the requirements are easily applicable to other cloud deployments. Anuket RM is implementation agnostic, viz., whether the implementation uses agents, federations or some other mechanisms. 

The following tables list some of the requirements for the Hybrid, Edge, and Multi cloud operator Platform (HEMP). These requirements are in addition to the requirements in other chapters of this RM. 

**HEMP General Requirements**

+-------------+--------------------------------------------------------+------------------------------------------------------------+
| Ref         | Requirement                                            | Definition/Note                                            |
+=============+========================================================+============================================================+

> | hem.gen.001 | HEMP shall use only published APIs in managing         | For example, to accomplish the example in `hem.gen.003`    |

Author: ```suggestion
| hem.gen.001 | HEMP should use only published APIs in managing        | For example, to accomplish the example in `hem.gen.003`    |
```

|             | component clouds				       | it will use the published APIs of the target cloud.        |
+-------------+--------------------------------------------------------+------------------------------------------------------------+

> | hem.gen.002 | HEMP shall publish all of the APIs used by any of      | For example, the provided GUI portal shall only use        |

Author: ```suggestion
| hem.gen.002 | HEMP should publish all of the APIs used by any of     | For example, the provided GUI portal shall only use        |
```

|             | its components                                         | HEMP published APIs                                        |
+-------------+--------------------------------------------------------+------------------------------------------------------------+

> | hem.gen.003 | HEMP shall provide for common terms for interaction    | For example, "create Account" across the different         |

Author: ```suggestion
| hem.gen.003 | HEMP should provide for common terms for interaction    | For example, "create Account" across the different         |
```

|             | with its constituent clouds                            | clouds                                                     |
+-------------+--------------------------------------------------------+------------------------------------------------------------+

> | hem.gen.004 | HEMP shall generalise and define a common set of       | Example resources: hosts (including BareMetal),            |

Author: ```suggestion
| hem.gen.004 | HEMP should generalise and define a common set of      | Example resources: hosts (including BareMetal),            |
```

|             | resources available to be managed inconstituent clouds | Virtual Machines (VM), vCPU, Memory, Storage, Network,     |
|             |                                                        | kubernetes clusters, kubernetes nodes, Images (OS, and     |
|             |                                                        | others), credentials. For private cloud additional         |
|             |                                                        | example resources: Racks, ToR/CE switches, Platform images |
+-------------+--------------------------------------------------------+------------------------------------------------------------+

> | hem.gen.005 | HEMP shall provide a common interface for managing     |                                                            |

Author: ```suggestion
| hem.gen.005 | HEMP should provide a common interface for managing    |                                                            |
```

|             | component clouds				       |                                                            |
+-------------+--------------------------------------------------------+------------------------------------------------------------+

> | hem.gen.006 | HEMP shall expose resources from all cloud operators   | See example of resources in `hem.gen.004` Definition/Note  |

Author: ```suggestion
| hem.gen.006 | HEMP should expose resources from all cloud operators   | See example of resources in `hem.gen.004` Definition/Note  |
```

> | hem.gen.006 | HEMP should expose resources from all cloud operators   | See example of resources in `hem.gen.004` Definition/Note  |

Author: ```suggestion
| hem.gen.006 | HEMP should expose resources from all cloud operators  | See example of resources in `hem.gen.004` Definition/Note  |
```

|             | and locations (regions, sites, etc.)	               |                                                            |
+-------------+--------------------------------------------------------+------------------------------------------------------------+

> | hem.gen.007 | HEMP shall allow reservation of resources if the       |                                                            |

Author: ```suggestion
| hem.gen.007 | HEMP should allow reservation of resources if the       |                                                            |
```

tools or APIs to create the necessary accounts, billing arrangements, quotas, etc. Then create the needed resources, 
such as VMs or Kubernetes clusters, etc. Following up with creating needed storage, networking, etc. before 
onboarding the workload and operating it. This is complex even when the operator is dealing with say only one other 
cloud, in addition to operating its own cloud. The operator is faced with a number of challenges including acquiring a 
new set of skills, knowledge of APIs, tools, and the complexity of managing different policies, updates, etc. This 
becomes impossible to manage when incorporating more than one other cloud. Hence the need for a Single Pane of Glass.

This Hybrid, Edge, and Multi-Cloud unified management Platform (HEMP) (a.k.a. Single-Pane-of-Glass) provides capabilities to consistently:

-  manage accounts, credentials, resources and services
  -  across facilities (regions, data centers, edge locations)
-  interoperate the different clouds
-  implement common policies and governance standards
-  manage a common security posture
-  provide an integrated visualisation into the infrastructure and workloads.

> through a common set of governance and operational practices.

Author: @joaoarodrigues The common set of governance and operational refers to the constituent clouds. So any capabilities provided by constituent clouds can be managed using common terms and methods rather than through the different terms and methods used by constituent cloud methods. It is not "common" for all "Teams".
